From f4e96c63ef633d5c271fe0893c707f272e9908ef Mon Sep 17 00:00:00 2001
From: "Al Sadi, Amir" <a.al-sadi@imperial.ac.uk>
Date: Tue, 21 Oct 2025 16:08:52 +0000
Subject: [PATCH 06/14] Delete keeptest

---
 keeptest | 1 -
 1 file changed, 1 deletion(-)
 delete mode 100644 keeptest

diff --git a/keeptest b/keeptest
deleted file mode 100644
index d00491fd7e5b..000000000000
--- a/keeptest
+++ /dev/null
@@ -1 +0,0 @@
-1
-- 
2.34.1

