From 99a994f9c56c8f4d6ed79220867c6777a4500865 Mon Sep 17 00:00:00 2001
From: Amir Al Sadi <alsadi.amir@gmail.com>
Date: Fri, 26 Dec 2025 19:33:07 +0000
Subject: [PATCH 14/14] removing PD at destroy time

---
 arch/arm64/kvm/rme.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/arch/arm64/kvm/rme.c b/arch/arm64/kvm/rme.c
index 90a96743efef..7f528521a7d6 100644
--- a/arch/arm64/kvm/rme.c
+++ b/arch/arm64/kvm/rme.c
@@ -546,6 +546,7 @@ static void realm_unmap_shared_range(struct kvm *kvm,
 		}
 
 		ret = rmi_rtt_unmap_unprotected(rd, addr, level, &next_addr);
+		//add diversion into unmap flow..but in RMM
 		switch (RMI_RETURN_STATUS(ret)) {
 		case RMI_SUCCESS:
 			break;
@@ -1770,16 +1771,6 @@ void kvm_destroy_realm(struct kvm *kvm)
 		realm->rd = NULL;
 	}
 
-	//destroying PD must be after destroying RD
-	if (realm->pd) {
-		phys_addr_t pd_phys = virt_to_phys(realm->pd);
-
-		if (WARN_ON(rmi_realm_destroy(pd_phys)))
-			return;
-		free_delegated_granule(pd_phys);
-		realm->pd = NULL;
-	}
-
 	rme_vmid_release(realm->vmid);
 
 	for (i = 0; i < pgd_size; i += RMM_PAGE_SIZE) {
@@ -1789,6 +1780,15 @@ void kvm_destroy_realm(struct kvm *kvm)
 			return;
 	}
 
+	//destroying PD must be after destroying everything else
+	if (realm->pd) {
+		phys_addr_t pd_phys = virt_to_phys(realm->pd);
+
+		if (WARN_ON(rmi_granule_undelegate(pd_phys)))
+			return;
+		realm->pd = NULL;
+	}
+
 	WRITE_ONCE(realm->state, REALM_STATE_DEAD);
 
 	/* Now that the Realm is destroyed, free the entry level RTTs */
-- 
2.34.1

