From 35b5fcf909891da6d4ce2fb013c47336233bb1d5 Mon Sep 17 00:00:00 2001
From: Amir Al Sadi <alsadi.amir@gmail.com>
Date: Mon, 10 Nov 2025 16:36:04 +0000
Subject: [PATCH 09/18] bugfixing policy config file

---
 drivers/misc/rsi_uploader.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/misc/rsi_uploader.c b/drivers/misc/rsi_uploader.c
index 26ff089994a1..8a5f5371fa85 100644
--- a/drivers/misc/rsi_uploader.c
+++ b/drivers/misc/rsi_uploader.c
@@ -361,7 +361,7 @@ static int __init rsi_policy_json_init(void)
 	vfree(json);
 
 	if (rc) {
-		pr_err("rsi_policy_json: failed to build binary from json (rc=%d)\n", rc);
+		pr_err("rsi_upload_policy: failed to build binary from json (rc=%d)\n", rc);
 		free_page((unsigned long)page);
 		return rc;
 	}
@@ -371,16 +371,16 @@ static int __init rsi_policy_json_init(void)
 	 * RAM window (identity-ish). If that’s the case, we can just pass
 	 * the page address as the IPA.
 	 */
-	ipa = (unsigned long)page;
+	ipa = virt_to_phys(page);
 
 	st = rsi_upload_policy(ipa);
 	if (st) {
-		pr_err("rsi_policy_json: RSI failed: 0x%lx\n", st);
+		pr_err("rsi_upload_policy: RSI failed: 0x%lx\n", st);
 		free_page((unsigned long)page);
 		return -EFAULT;
 	}
 
-	pr_info("rsi_policy_json: sent config to RMM (ipa=0x%lx, bin_len=%zu)\n",
+	pr_info("rsi_upload_policy: sent config to RMM (ipa=0x%lx, bin_len=%zu)\n",
 		ipa, bin_len);
 
 	/* keep the page around if the RMM expects to read again; otherwise free it */
-- 
2.34.1

