From 97d956ffe0037ea1c2802a77362717a59c6798c2 Mon Sep 17 00:00:00 2001
From: Amir Al Sadi <alsadi.amir@gmail.com>
Date: Wed, 24 Dec 2025 15:37:12 +0000
Subject: [PATCH 16/18] minor fix to rsi uploader

---
 drivers/misc/rsi_uploader.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/rsi_uploader.c b/drivers/misc/rsi_uploader.c
index ddaafa059342..776d1c647e20 100644
--- a/drivers/misc/rsi_uploader.c
+++ b/drivers/misc/rsi_uploader.c
@@ -1176,10 +1176,20 @@ int rmm_payload_to_json(const unsigned char *buf, size_t len,
             char map_vm_name[5];
             const char *prot_str = prot_to_str(prot);
 
-            if (vm_idx < num_vms)
+            // if (vm_idx < num_vms)
+            //     name4_to_cstr(vms[vm_idx].name4, map_vm_name);
+            // else
+            //     snprintf(map_vm_name, sizeof(map_vm_name), "VM%u", vm_idx);
+
+            /* Resolve mapping VM name */
+            if (vm_idx == VM_IDX_ANY) {
+                snprintf(map_vm_name, sizeof(map_vm_name), "ANY");
+            } else if (vm_idx < num_vms) {
                 name4_to_cstr(vms[vm_idx].name4, map_vm_name);
-            else
+            } else {
                 snprintf(map_vm_name, sizeof(map_vm_name), "VM%u", vm_idx);
+            }
+
 
             APPEND("      \"%s\": { \"gpa\": 0x%llx, \"prot\": \"%s\" }%s\n",
                    map_vm_name,
-- 
2.34.1

